### PR TITLE
[RPCRT4] Sync msvc.S with WINE's inline assembly,

### PR DESCRIPTION
**Pull request by learn-more**
as suggested by Thomas

[CORE-18505](https://jira.reactos.org/browse/CORE-18505)

## Purpose

This fixes the VirtualBox 2.0.0 installer.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
Approval given based on positive test outcome. Not from understanding the ASM changes.

**Comment by JoachimHenze** [comment]
Do we need something similar also for MSVC x64?

**Review by ThFabba — APPROVED** [review]

**Comment by learn-more** [comment]
> Do we need something similar also for MSVC x64?

Probably, but I don't have a testcase nor a working gcc x64, and I don't fancy spending a lot of time deciphering that AT&T syntax.

File: dll/win32/rpcrt4/msvc.S [hunk 1]
```diff
@@ -13,24 +13,25 @@ EXTERN _ndr_client_call:PROC
...
+    shr eax, 1
+    movzx eax, word ptr [edx+eax+4] /* arguments size */
     push eax
-    lea eax, [esp+8]       /* &This */
+    lea eax, [esp+8]                /* &This */
     push eax
-    push edx               /* format string */
-    push [ecx]             /* info->pstubdesc */
+    push edx                        /* format string */
+    push [ecx]                      /* info->pstubdesc */
```
Comment (tkreuzer) [inline dll/win32/rpcrt4/msvc.S:30]: You might want to add a `dword ptr` here. Not sure it it's needed though.

**Review by tkreuzer — APPROVED** [review]

**Comment by julenuri** [comment]
I have a testing GCC x64 (not of current master). It's not based on the current GCC 8 that we use, but GCC 11, that have few hacks to load (made by Simone Lombardo). 
I don't believe RPCRT4 works inside of it, but I had to retest - since there are VBox 2.0.0 XP64 / amd64 versions.

**Comment by julenuri** [comment]
Fixed for me under x86 MSVC, so the ticket might be closed as merged :) Thanks you
Captura desde 2022-11-26 19-46-17
.

**Review by binarymaster — APPROVED** [review]

**Pull request merged**